client/my-sites/checkout/upsell-nudge/professional-email-upsell/index.tsx

```diff
@@ -72,29 +87,64 @@ const ProfessionalEmailUpsell = ( {
 
 	const optionalMailboxFields = [ 'alternativeEmail', 'name' ];
 
+	const changeIntervalLength = ( newIntervalLength: IntervalLength ) => {
+		setSelectedIntervalLength( newIntervalLength );
+		dispatch(
+			recordTracksEvent( 'calypso_professional_email_upsell_nudge_billing_interval_toggle_click', {
+				domain_name: domainName,
+				new_interval: newIntervalLength,
+			} )
+		);
+	};
+
+	/**
+	 * We calculate the price for a year subscription, given how many months we are going to offer for free.
+	 * It takes into account leap years, and months like february.
+	 *
+	 * Example: If we give 3 months for free in February, for a product that cost 35$ yearly then we return the price
+	 * that we need to bill after those 3 months for the rest of the year.
+	 *
+	 * @param productCost
+	 * @param freeMonths
+	 */
+	const getProratedPrice = ( productCost: number, freeMonths: number ) => {
+		const now = new Date();
+		const freeDays = new Date( new Date().setMonth( now.getMonth() + freeMonths ) ).getTime();
+		const nextYearDate = new Date( new Date().setFullYear( now.getFullYear() + 1 ) );
+		const diff = new Date( nextYearDate.getTime() - freeDays );
+		const diffDays =
+			( nextYearDate.getTime() - now.getTime() ) / 86400000 - diff.getTime() / 86400000;
+		const price = ( 365 - diffDays ) / 365;
+		return price * productCost;
+	};
```

Comment:
I have several comments:
1. This should be explicitly identified as applying to annual subscriptions, e.g. `getProratedAnnualPrice()`
3. This may make sense as a standalone function in `client/lib/titan`, or even as a function declared outside the context of `ProfessionalEmailUpsell` -- this code doesn't depend on anything other than the input arguments.
4. Nit: we should declare a return type of `number`.
5. We may want to consider rounding the returned value to 2 decimal digits
```suggestion
	/**
	 * We calculate the price for a year subscription, given how many months we are going to offer for free.
	 * It takes into account leap years, and months like february.
	 *
	 * Example: If we give 3 months for free in February, for a product that cost 35$ yearly then we return the price
	 * that we need to bill after those 3 months for the rest of the year.
	 *
	 * @param productCost
	 * @param freeMonths
	 */
	const getProratedAnnualPrice = ( productCost: number, freeMonths: number ) => {
		const now = new Date();
		const freeDays = new Date( new Date().setMonth( now.getMonth() + freeMonths ) ).getTime();
		const nextYearDate = new Date( new Date().setFullYear( now.getFullYear() + 1 ) );
		const diff = new Date( nextYearDate.getTime() - freeDays );
		const diffDays =
			( nextYearDate.getTime() - now.getTime() ) / 86400000 - diff.getTime() / 86400000;
		const price = ( 365 - diffDays ) / 365;
		return Math.round( 100 * price * productCost ) / 100;
	};
```